File: backend/models/product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: [true, 'Product name is required'],
      trim: true,
      minlength: [3, 'Name must be at least 3 characters'],
      maxlength: [100, 'Name cannot exceed 100 characters']
    },
    description: {
      type: String,
      required: [true, 'Product description is required'],
      trim: true,
      maxlength: [1000, 'Description cannot exceed 1000 characters']
    },
    cost: {
      type: Number
      
    },
    category: {
      type: String,
      enum: ['Fruit', 'Vegetable', 'Dairy', 'Grain', 'Other'],
      default: 'Other'
    },
    inStock: {
      type: Boolean,
      default: true
    },
    quantity: {
      type: Number,
      default: 0,
      min: [0, 'Quantity must be at least 0']
    }
  },
  {
    timestamps: true // Automatically adds createdAt and updatedAt
  }
);

module.exports = mongoose.model('Product', productSchema);

File: frontend/lib/models/product.dart


class Product {
  final String id;
  final String name;
  final String description;
  final double cost;
  final String category;
  final bool inStock;
  final int quantity;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.cost,
    required this.category,
    required this.inStock,
    required this.quantity,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      cost: (json['cost'] as num).toDouble(), // Safely handles both int & double
      category: json['category'],
      inStock: json['inStock'],
      quantity: json['quantity'],
    );
  }
}

